File: packages/backend/src/main/kotlin/hollybike/api/types/user/EUserStatus.kt
/*
  Hollybike API Kotlin KTor Graalvm application
  Made by MacaronFR (Denis TURBIEZ) and Loïc Vanden Bossche
*/
package hollybike.api.types.user

import kotlinx.serialization.Serializable

@Serializable
enum class EUserStatus(val value: Int) {
	Enabled(1),
	Disabled(-1);

	companion object {
		operator fun get(value: Int) = entries.first { it.value == value }
	}
}

File: packages/frontend/src/types/EUserStatus.ts
/*
  Hollybike Back-office web application
  Made by MacaronFR (Denis TURBIEZ) and enzoSoa (Enzo SOARES)
*/
import { Option } from "../components/Select/Select.tsx";

export enum EUserStatus {
	Enabled = "Enabled",
	Disabled = "Disabled"
}

export const EUserStatusOptions: Option[] = [
	{
		name: "Activé",
		value: EUserStatus.Enabled,
	},
	{
		name: "Désactivé",
		value: EUserStatus.Disabled,
	},
];

export function EUserStatusToString(v: EUserStatus) {
	switch (v) {
	case EUserStatus.Enabled: return "Activé";
	case EUserStatus.Disabled: return "Désactivé";
	}
}
